### ISSUE TITLE
初学者求lstm+cnn/tensorflow代码？｜New beginner for "lstm+cnn/tensorflow"

### ISSUE THREAD

**Issue by WaveLi123** [white127/insuranceQA-cnn-lstm#3]

**Comment by pcgreat** [comment]
这是我试着写的，希望能帮到你https://github.com/pcgreat/insuranceQA-cnn-lstm/tree/master/lstm_cnn/tensorflow

**Comment by chenzhi1992** [comment]
tf版本的lstm+cnn的代码,怎么还是cnn的代码啊?@ pcgreat

**Comment by pcgreat** [comment]
@chenzhi1992 抱歉，上传代码的时候弄错了，现在更正了

**Comment by chenzhi1992** [comment]
谢谢回复！


发自网易邮箱大师
在2017年07月21日 23:55，Chao Pan 写道:

@chenzhi1992 抱歉，上传代码的时候弄错了，现在更正了

**Comment by chenzhi1992** [comment]
你的lstm+cnn 大概训练多少step后loss收敛?我训练到1000step,acc一直在0.1到0.2之间.@pcgreat 
@ pcgreat

**Comment by pcgreat** [comment]
1000步太少了，至少12k步才能看出点效果。你可以适当加一点优化，比如有选择地抽取negative的sample，这样收敛快一些

**Comment by zifengwen** [comment]
cnn/tensorflow/insqa_cnn  中：
        with tf.name_scope("accuracy"):
            self.correct = tf.equal(zero, self.losses)
            self.accuracy = tf.reduce_mean(tf.cast(self.correct, "float"), name="accuracy")
这几行不足以评估模型的好坏吧，只是统计训练数据中“一个正确答案跟一个错误答案”选对正确答案的比例。。

**Comment by zifengwen** [comment]
cnn/tensorflow/insqa_cnn  中：
        with tf.name_scope("accuracy"):
            self.correct = tf.equal(zero, self.losses)
            self.accuracy = tf.reduce_mean(tf.cast(self.correct, "float"), name="accuracy")
这几行不足以评估模型的好坏吧，只是统计训练数据中“一个正确答案跟一个错误答案”选对正确答案的比例。。

**Comment by zifengwen** [comment]
cnn/tensorflow/insqa_train 中，模型评估的方法dev_step（）中，实际上只评估了20条数据吧。
为什么不是直接所有答案的向量计算出来，，然后对于新的问题，计算向量，匹配所有答案向量，得到最佳答案，跟真实的答案比较得到准确率呢？？
现在这样写有什么好处？？
